source-mysql/replication.go

```diff
@@ -523,203 +504,234 @@ func (rs *mysqlReplicationStream) handleRowsEvent(ctx context.Context, event *re
 		return fmt.Errorf("unknown key columns for stream %q", streamID)
 	}
 
-	var nonTransactionalTable = rs.isNonTransactional(streamID)
+	// Construct efficient indices and transcoder arrays for value processing
+	//
+	// TODO(wgd): Hoist this to a higher level
```

Comment:
Also noting this TODO, which seems like it may be more relevant to address now, although I'm not 100% sure if that was the intention.